extensions/data-plane/data-plane-azure-storage/src/main/java/org/eclipse/edc/connector/dataplane/azure/storage/pipeline/AzureStorageDataSource.java

```diff
@@ -48,14 +54,24 @@ private AzureStorageDataSource() {
 
     @Override
     public StreamResult<Stream<Part>> openPartStream() {
-        return success(Stream.of(getPart()));
+
+        if (!StringUtils.isNullOrEmpty(blobPrefix)) {
+            var folderBlobs = blobStoreApi.listContainerFolder(accountName, containerName, blobPrefix);
+            if (folderBlobs.isEmpty()) {
+                monitor.severe(format("Error listing blobs in the container %s with prefix %s", containerName, blobPrefix));
```

Comment:
I don't think this is actually a failure. Attempting to copy the contents of an empty folder are simply a NOOP, and I would return an empty stream.
I would only return a `failure` if the actual listing operation did not work.

Reply:
I tried to kept consistency with the other extension - AWS - which already supports multi objects transfer - There it is consider as failure 
https://github.com/eclipse-edc/Technology-Aws/blob/main/extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSource.java - line 53

Reply:
ok, that makes sense.